scripts/content-utils/card-tree.ts

```diff
@@ -21,7 +21,131 @@ export type CardLeaf = {
     modifiers?: GameWorldModifier | GameWorldModifier[]
 }
 
+/**
+ *
+ * @param tree The CardTree to recursively turn into an array of cards
+ *
+ * NOTE: The following params should only be used internally by the content utils
+ * to connect CardTrees within other structures, like for example CardSequences.
+ *
+ * For actual content, use `tree.conditions` and `tree.modifiers` instead:
+ * @param _startConditions Internal WorldQuery[] to control when to show this CardTree node. Useful when you want to complete the previous node in a CardSequence before starting the CardTree.
+ * @param _endModifiers Internal GameWorldModifier[] to apply to all CardTree leaf nodes. Useful when you want to complete a CardTree within a CardSequence before moving on to the next node in the CardSequence.
+ */
 export function cardsFromTree(
+    tree: Omit<CardTree, "modifiers">,
+    _startConditions?: WorldQuery[],
+    _endModifiers: GameWorldModifier[] = [],
+): CardData[] {
+    // HACK: retrieving the bindRef from the first _startConditions is hack
+    // that should be improved to ensure we get the correct reference in case we have multiple _startConditions.
+
+    // IDEA: Possibly we should update types for _startConditions and _endModifiers to only ever contain one item
+    // This would reduce complexity of `cardsFromTree()`, while still providing us with most of the same functionality
+    // Since these are only supposed to be used internally, I currently don't see a good use case to support multiple _startConditions or _endModifiers.
+
+    const bindRef = _startConditions?.length
+        ? _startConditions[0]["flags"] &&
+          Object.keys(_startConditions[0]["flags"])[0]
+        : undefined
```

Comment:
This solution is breaking `CardTrees` nested within `CardSequences`, since the `flag` pulled from `_startConditions` will effectively be overwritten by the `CardTree` logic, which breaks the `CardSequence`.

One possible solution might be to switch back to passing down the `bindRef` as a fourth, internal argument to `cardsFromTree()`. I'll try that and see if it solves the problem

Reply:
Solved this by passing `bindRef` as the fouth internal argument. This also solved another problem that I discovered while testing the first. Big win! 🎉